### PR TITLE
feat: Windows table setup

### PR DESCRIPTION
**Pull request by yahu1031**
Currently, we don't have the support for setting up pgsql tables automation using a script file for windows. I have made that enhancement.

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`), and made sure that the document follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [x] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

## Breaking changes

No breaking changes

## Enhancement

Setting up tables in windows was added using a batch file.

### CODE REVIEW

File: tools/serverpod_cli/bin/util/constants.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+/// Health check command
+const cmdHealthCheck = 'healthcheck';
+
+/// Pubspecs generate command
+const cmdGeneratePubspecs = 'generate-pubspecs';
+
+/// Serverpod run modes.
+final runModes = <String>['development', 'staging', 'production'];
+
+/// Getter for the Windows platform.
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/util/constants.dart:42]: This just redefine already defined constants. This makes the code less readable.

File: tools/serverpod_cli/pubspec.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ dependencies:
   recase: ^4.0.0
   super_string: ^1.0.3
   watcher: ^1.0.0
-  analyzer: ^3.3.1
```
Comment (vlidholt) [inline tools/serverpod_cli/pubspec.yaml:22]: This should not be part of this PR

File: tools/serverpod_cli/bin/util/command_line_tools.dart [hunk 3]
```diff
@@ -82,7 +89,7 @@ class _CommandFormatter {
   late final List<String> args;
 
   _CommandFormatter(String command, this.args) {
-    this.command = Platform.isWindows ? '$command.bat' : command;
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/util/command_line_tools.dart:85]: Is this not breaking commands other than the script? E.g. `flutter create`?

File: templates/serverpod_templates/projectname_server/setup-tables.bat [hunk 4]
```diff
@@ -0,0 +1,22 @@
+@echo off
```
Comment (vlidholt) [inline templates/serverpod_templates/projectname_server/setup-tables.bat:1]: This file looks good! If you make a change that only adds this file, I can approve it and fix the other stuff around it.

File: tools/serverpod_cli/bin/create/create.dart [hunk 5]
```diff
@@ -354,18 +355,18 @@ Future<void> performCreate(
...
     printww('You are almost ready to rock!');
     printwwln('To get going, you need to start Docker by running:');
     stdout.writeln('  \$ cd ${p.join(name, '${name}_server')}');
     stdout.writeln('  \$ docker-compose up --build --detach');
-    printww('');
-    printwwln(
-        'When your docker container is up and running you need to install the default Serverpod postgres tables. (You only need to to this once.)');
-    stdout.writeln(
-        '  \$ Get-Content .\\generated\\tables-serverpod.pgsql | docker-compose run -T postgres env PGPASSWORD="$dbPassword" psql -h postgres -U postgres -d $name');
+    // printww('');
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/create/create.dart:365]: Never include commented out code in PRs generally.

File: tools/serverpod_cli/bin/serverpod_cli.dart [hunk 6]
```diff
@@ -14,27 +14,13 @@ import 'internal_tools/generate_pubspecs.dart';
 import 'run/runner.dart';
 import 'shared/environment.dart';
 import 'util/command_line_tools.dart';
+import 'util/constants.dart';
 import 'util/print.dart';
 
-const cmdCreate = 'create';
```
Comment (vlidholt) [inline tools/serverpod_cli/bin/serverpod_cli.dart:19]: Breaking out the constants shouldn't be part of this PR. Not sure I think it improves readability of the app either as it is would break up where you need to look for the constants.

**Review by vlidholt — CHANGES_REQUESTED** [review]
There are several changes that shouldn't be part of this PR and a potential bug, so I cannot approve it as is.

However, the bat files look good, so if you make another PR with just that file I can approve it.

File: templates/serverpod_templates/projectname_server/setup-tables.bat (hunk 4 above)
Comment (yahu1031) [reply templates/serverpod_templates/projectname_server/setup-tables.bat:1]: Okie I will revert the code to previous one.

File: tools/serverpod_cli/bin/util/command_line_tools.dart (hunk 3 above)
Comment (yahu1031) [reply tools/serverpod_cli/bin/util/command_line_tools.dart:85]: okie.

File: tools/serverpod_cli/bin/util/constants.dart (hunk 1 above)
Comment (yahu1031) [reply tools/serverpod_cli/bin/util/constants.dart:42]: reverting it.

**Review by vlidholt — APPROVED** [review]
Great! Looking good, this is what a PR should look like. :) Clear, concise, and only making the changes needed to fix the issue.
